functests/gatekeeper/gatekeeper.go

```diff
@@ -0,0 +1,637 @@
+				kind:  "annotations",
+				key:   "changed",
+				value: "true",
+			},
+		}...)...,
+	)
+
+	It("should apply mutations by order", func() {
+		amDefB := &assignMetadataInfo{
+			name:  "mutation-b",
+			kinds: []string{"Pod"},
+			scope: "Namespaced",
+			metadata: metadataInfo{
+				kind:  "labels",
+				key:   "mutated-by",
+				value: "b",
+			},
+		}
+
+		By("Creating assignMetadata mutation-b")
+		amB := assignMetadataDefinition(amDefB)
+		amB, err := client.Resource("assignmetadata").Create(context.Background(), amB, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		defer func() {
+			err := client.Resource("assignmetadata").Delete(context.Background(), amB.GetName(), metav1.DeleteOptions{})
+			Expect(err).ToNot(HaveOccurred())
+		}()
+
+		By("Creating test-pod-b")
+		testPodB := GetTestPod(TestingNamespace)
+		testPodB.SetName("test-pod-b")
+		testPodB.Unstructured, err = client.Resource("pods").Namespace(testPodB.GetNamespace()).Create(context.Background(), testPodB.Unstructured, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		defer func() {
+			err := client.Resource("pods").Namespace(testPodB.GetNamespace()).Delete(context.Background(), testPodB.GetName(), metav1.DeleteOptions{})
+			Expect(err).ToNot(HaveOccurred())
+		}()
+
+		By("Asserting that test-pod-b was labeled")
+		value, ok, err := unstructured.NestedString(testPodB.Object, "metadata", amDefB.metadata.kind, amDefB.metadata.key)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(ok).To(BeTrue())
+		Expect(value).To(Equal(amDefB.metadata.value))
+
+		amDefA := &assignMetadataInfo{
+			name:  "mutation-a",
+			kinds: []string{"Pod"},
+			scope: "Namespaced",
+			metadata: metadataInfo{
+				kind:  "labels",
+				key:   "mutated-by",
+				value: "a",
+			},
+		}
+
+		By("Creating assignMetadata mutation-a")
+		amA := assignMetadataDefinition(amDefA)
+		amA, err = client.Resource("assignmetadata").Create(context.Background(), amA, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		defer func() {
+			err := client.Resource("assignmetadata").Delete(context.Background(), amA.GetName(), metav1.DeleteOptions{})
+			Expect(err).ToNot(HaveOccurred())
+		}()
+
+		By("Creating test-pod-a")
+		testPodA := GetTestPod(TestingNamespace)
+		testPodA.SetName("test-pod-a")
+		testPodA.Unstructured, err = client.Resource("pods").Namespace(testPodA.GetNamespace()).Create(context.Background(), testPodA.Unstructured, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		defer func() {
+			err := client.Resource("pods").Namespace(testPodA.GetNamespace()).Delete(context.Background(), testPodA.GetName(), metav1.DeleteOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+		}()
+
+		By("Asserting that test-pod-a was labeled")
+		value, ok, err = unstructured.NestedString(testPodA.Object, "metadata", amDefA.metadata.kind, amDefA.metadata.key)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(ok).To(BeTrue())
+		Expect(value).To(Equal(amDefA.metadata.value))
```

Comment:
This test could pass if mutators are not ordered always the same, but ordered by order of adding them.
Maybe we can check if adding first A then B also produces the same result, or add a 3rd mutator which would be applied last, but not be applied before mutation-a.

The test would like like:

```
apply mutation-b 
create pod 
check mutation-b is applied

apply mutation-a
create pod 
check mutation-a is applied

apply mutation-c
create pod 
check mutation-a is applied (NOT C)
```
For mutator ordering check:
https://github.com/open-policy-agent/gatekeeper/blob/master/pkg/mutation/assignmeta_mutator.go#L69
https://github.com/open-policy-agent/gatekeeper/blob/master/pkg/mutation/assign_mutator.go#L55